lua/telescope/_extensions/project.lua

```diff
@@ -9,54 +9,59 @@ local finders = require("telescope.finders")
 local pickers = require("telescope.pickers")
 local conf = require("telescope.config").values
 local builtin = require("telescope.builtin")
-local project_dirs_file = vim.fn.stdpath('data') .. '/telescope-projects.lua'
+-- format for projects is "title of project"="~/this/path/name"
+local project_dirs_file = vim.fn.stdpath('data') .. '/telescope-projects.txt'
 
 local select_project = function(opts, projects, run_task_on_selected_project)
-	pickers.new(opts, {
-		prompt_title = 'Select a project',
-		results_title = 'Projects',
-		finder = finders.new_table {results = projects},
-		sorter = conf.file_sorter(opts),
-		attach_mappings = function(prompt_bufnr)
-			local on_project_selected = function()
-				local selection = actions.get_selected_entry(prompt_bufnr)
-				actions.close(prompt_bufnr)
-				run_task_on_selected_project(selection.value)
-			end
-			actions.goto_file_selection_edit:replace(on_project_selected)
-			return true
-		end
-	}):find()
+	print(vim.inspect(projects))
+	-- pickers.new(opts, {
+	-- 	prompt_title = 'Select a project',
+	-- 	results_title = 'Projects',
+	-- 	finder = finders.new_table {results = projects},
+	-- 	sorter = conf.file_sorter(opts),
+	-- 	attach_mappings = function(prompt_bufnr)
+	-- 		local on_project_selected = function()
+	-- 			local selection = actions.get_selected_entry(prompt_bufnr)
+	-- 			actions.close(prompt_bufnr)
+	-- 			run_task_on_selected_project(selection)
+	-- 		end
+	-- 		actions.goto_file_selection_edit:replace(on_project_selected)
+	-- 		return true
+	-- 	end
+	-- }):find()
 end
 
 local project = function(opts)
 	opts = opts or {}
 
-	-- TODO figure this out:
-
-	-- local project_dirs = {}
-	-- for line in io.lines(project_dirs_file) do
-	-- 	table.insert(project_dirs, line)
-	-- end
+	local project_dirs = {}
+	for line in io.lines(project_dirs_file) do
+		local title = (line:match '[^"]+')
+		local path = (line:match [[="([^"]+)]])
```

Comment:
You can do this in one match. Get rid of the other parentheses. So `local title, path =  line:match('(title matching regex)%s(path matching regex)')`. It will return what is matched inside the parentheses. %s it he separator here space